# Project 3

In [39]:
import plotly.io as pio

pio.renderers.default = "vscode+jupyterlab+notebook_connected"

In [40]:
import pandas as pd
import plotly.express as px

In [41]:
df = pd.read_csv('/Users/yutonomy/Dropbox/Columbia SIPA/24FALL/Computing in Context/project3/Neighborhood.csv')
df

Year Published  PUMA Borough                               Neighborhoods  \
0              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
1              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
2              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
3              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
4              2020  3701   Bronx          Riverdale, Fieldston & Kingsbridge   
..              ...   ...     ...                                         ...   
380            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
381            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
382            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
383            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   
384            2020  4114  Queens  Far Rockaway, Breezy Point & Broad Channel   

                           CD  Join  NYC_Poverty_Rate  Median_Income  \
0     BX Community District 8  3701             0.152          30437   
1     BX Community District 8  3701             0.152          30437   
2     BX Community District 8  3701             0.152          30437   
3     BX Community District 8  3701             0.152          30437   
4     BX Community District 8  3701             0.152          30437   
..                        ...   ...               ...            ...   
380  QN Community District 14  4114             0.179          25920   
381  QN Community District 14  4114             0.179          25920   
382  QN Community District 14  4114             0.179          25920   
383  QN Community District 14  4114             0.179          25920   
384  QN Community District 14  4114             0.179          25920   

     Perc_White  Perc_Black  ...  Ind6Rank  \
0         0.324       0.124  ...      32.0   
1         0.324       0.124  ...      38.0   
2         0.324       0.124  ...      27.0   
3         0.324       0.124  ...      41.0   
4         0.324       0.124  ...      28.0   
..          ...         ...  ...       ...   
380       0.350       0.353  ...      42.0   
381       0.350       0.353  ...      55.0   
382       0.350       0.353  ...      15.0   
383       0.350       0.353  ...       NaN   
384       0.350       0.353  ...       NaN   

                                        Ind6Definition  \
0    Percentage of jobs in small businesses (less t...   
1    Percentage of bank & credit union branches off...   
2    Number of pre-foreclosure notices issued per 1...   
3        Average (mean) travel time to work in minutes   
4    Percentage of people eligible for SNAP receivi...   
..                                                 ...   
380  Number of pre-foreclosure notices issued per 1...   
381      Average (mean) travel time to work in minutes   
382  Percentage of people eligible for SNAP receivi...   
383                                                NaN   
384                                                NaN   

                           Ind7 Ind7Outcome Ind7Rank  \
0       Participatory Budgeting    0.071777     26.0   
1    Mobile Banking Utilization    0.205800     23.0   
2          Financial Counseling    0.000000     40.0   
3                           NaN         NaN      NaN   
4        Tax Credit Utilization    0.966000      9.0   
..                          ...         ...      ...   
380        Financial Counseling    1.413969     21.0   
381                         NaN         NaN      NaN   
382      Tax Credit Utilization    0.936000     25.0   
383                         NaN         NaN      NaN   
384                         NaN         NaN      NaN   

                                        Ind7Definition  \
0    Percentage of eligible residents casting a vot...   
1    Estimated percentage of residents that used mo...   
2    Full-Time Equivalent Financial Counselors per ...   
3           

- **Dataset(s) to be used:** [https://data.cityofnewyork.us/Business/Neighborhood-Financial-Health-Digital-Mapping-and-/r3dx-pew9/about_data]
- **Analysis question:** The relationship between race and average income: is there any relationship between the percentage of black people and the change in the average income?
- **Columns that will (likely) be used:**
  - Median_Income
  - Perc_Black
- **Hypothesis**: The bigger the percentage of the black people gets, the less the average income becomes
- **Site URL:** [https://github.com/yutonomy/project3]

# Check the data and clean the data if needed

In [42]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 385 entries, 0 to 384
Data columns (total 52 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Year Published    385 non-null    int64  
 1   PUMA              385 non-null    int64  
 2   Borough           385 non-null    object 
 3   Neighborhoods     385 non-null    object 
 4   CD                385 non-null    object 
 5   Join              385 non-null    int64  
 6   NYC_Poverty_Rate  385 non-null    float64
 7   Median_Income     385 non-null    int64  
 8   Perc_White        385 non-null    float64
 9   Perc_Black        385 non-null    float64
 10  Perc_Asian        385 non-null    float64
 11  Perc_Other        385 non-null    float64
 12  Perc_Hispanic     385 non-null    float64
 13  Goal              385 non-null    object 
 14  GoalName          385 non-null    object 
 15  GoalFullName      385 non-null    object 
 16  TotalOutcome      367 non-null    float64
 1

In [43]:
df.head()

Year Published  PUMA Borough                       Neighborhoods  \
0            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
1            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
2            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
3            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   
4            2020  3701   Bronx  Riverdale, Fieldston & Kingsbridge   

                        CD  Join  NYC_Poverty_Rate  Median_Income  Perc_White  \
0  BX Community District 8  3701             0.152          30437       0.324   
1  BX Community District 8  3701             0.152          30437       0.324   
2  BX Community District 8  3701             0.152          30437       0.324   
3  BX Community District 8  3701             0.152          30437       0.324   
4  BX Community District 8  3701             0.152          30437       0.324   

   Perc_Black  ...  Ind6Rank  \
0       0.124  ...      32.0   
1       0.124  ...      38.0   
2       0.124  ...      27.0   
3       0.124  ...      41.0   
4       0.124  ...      28.0   

                                      Ind6Definition  \
0  Percentage of jobs in small businesses (less t...   
1  Percentage of bank & credit union branches off...   
2  Number of pre-foreclosure notices issued per 1...   
3      Average (mean) travel time to work in minutes   
4  Percentage of people eligible for SNAP receivi...   

                         Ind7 Ind7Outcome Ind7Rank  \
0     Participatory Budgeting    0.071777     26.0   
1  Mobile Banking Utilization    0.205800     23.0   
2        Financial Counseling    0.000000     40.0   
3                         NaN         NaN      NaN   
4      Tax Credit Utilization    0.966000      9.0   

                                      Ind7Definition  \
0  Percentage of eligible residents casting a vot...   
1  Estimated percentage of residents that used mo...   
2  Full-Time Equivalent Financial Counselors per ...   
3                                                NaN   
4     Percentage of eligible Tax Units claiming EITC   

                              Ind8 Ind8Outcome  Ind8Rank  \
0                              NaN         NaN       NaN   
1                              NaN         NaN       NaN   
2                              NaN         NaN       NaN   
3                              NaN         NaN       NaN   
4  Adults With High School Diploma    0.845032      21.0   

                                      Ind8Definition  
0                                                NaN  
1                                                NaN  
2                                                NaN  
3                                                NaN  
4  Percentage of adults with a high school degree...  

[5 rows x 52 columns]

In [44]:
df.sample(10)

Year Published  PUMA    Borough  \
374            2020  4113     Queens   
383            2020  4114     Queens   
125            2020  3807  Manhattan   
342            2020  4108     Queens   
376            2020  4113     Queens   
107            2020  3805  Manhattan   
252            2020  4012   Brooklyn   
194            2020  4004   Brooklyn   
175            2020  4001   Brooklyn   
49             2020  3707      Bronx   

                                    Neighborhoods  \
374                     Howard Beach & Ozone Park   
383    Far Rockaway, Breezy Point & Broad Channel   
125  Chelsea, Clinton & Midtown Business District   
342                      Forest Hills & Rego Park   
376                     Howard Beach & Ozone Park   
107                               Upper East Side   
252                 Sunset Park & Windsor Terrace   
194                Brooklyn Heights & Fort Greene   
175                     Greenpoint & Williamsburg   
49     Morris Heights, Fordham South & Mount Hope   

                               CD  Join  NYC_Poverty_Rate  Median_Income  \
374      QN Community District 10  4113             0.189          27016   
383      QN Community District 14  4114             0.179          25920   
125  MN Community Districts 4 & 5  3807             0.108          65905   
342       QN Community District 6  4108             0.163          41029   
376      QN Community District 10  4113             0.189          27016   
107       MN Community District 8  3805             0.073          71933   
252       BK Community District 7  4012             0.294          20787   
194       BK Community District 2  4004             0.121          44313   
175       BK Community District 1  4001             0.173          35889   
49        BX Community District 5  3707             0.344          15628   

     Perc_White  Perc_Black  ...  Ind6Rank  \
374       0.216       0.151  ...      22.0   
383       0.350       0.353  ...       NaN   
125       0.609       0.054  ...       NaN   
342       0.512       0.029  ...      41.0   
376       0.216       0.151  ...      43.0   
107       0.761       0.028  ...       4.0   
252       0.223       0.026  ...       NaN   
194       0.447       0.261  ...      13.0   
175       0.631       0.042  ...       2.0   
49        0.013       0.274  ...       NaN   

                                        Ind6Definition  \
374  Percentage of bank & credit union branches off...   
383                                                NaN   
125                                                NaN   
342  Percentage of people eligible for SNAP receivi...   
376      Average (mean) travel time to work in minutes   
107      Average (mean) travel time to work in minutes   
252                                                NaN   
194  Number of pre-foreclosure notices issued per 1...   
175  Percentage of people eligible for SNAP receivi...   
49                                                 NaN   

                           Ind7 Ind7Outcome Ind7Rank  \
374  Mobile Banking Utilization    0.224400     16.0   
383                         NaN         NaN      NaN   
125                         NaN         NaN      NaN   
342      Tax Credit Utilization    0.917000     36.0   
376                         NaN         NaN      NaN   
107                         NaN         NaN      NaN   
252                         NaN         NaN      NaN   
194        Financial Counseling    7.997022      2.0   
175      Tax Credit Utilization    0.946000     21.0   
49                          NaN         NaN      NaN   

                                        Ind7Definition  \
374  Estimated percentage of residents that used mo...   
383                                                NaN   
125                                                NaN   
342     Percentage of eligible Tax Units claiming EITC   
376                                                NaN   
107                             

# Data Visualization
We will do the data visualization: visualize in the bar graph and the scatterplot then figure out the relationship or correlation between % of black people and Median Incomes.

In [45]:
fig = px.bar(df, x='Perc_Black', y='Median_Income', color_discrete_sequence=['#000000'])  # Pure black color 
fig.show()


In [46]:
fig = px.scatter(df, x='Perc_Black', y='Median_Income')
fig.show()

# Reflection
Based on the bar graph and the scatterplot above, it can be said that the higher the percentage of Black people, the lower the median income tends to be, and conversely, the lower the percentage, the higher the median income.

# Check the regression

In [47]:
import numpy as np
x = df['Perc_Black']
y = df['Median_Income']

coefficients = np.polyfit(x, y, 1)

print(f"Regression coefficient (slope): {coefficients[0]}")
print(f"Intercept: {coefficients[1]}")

Regression coefficient (slope): -20276.573844372164
Intercept: 35800.935109920705


# Takeaway
The regression coefficient (slope): -20276.57 indicates that for every 1% increase in Perc_Black (percentage of the Black population), the Median_Income (median income) decreases by approximately $20,276. Since the value is negative, it suggests a negative linear relationship between Perc_Black and Median_Income.